### PR TITLE
Docker evaluation infrastructure

### PR DESCRIPTION
**Pull request by thomasyu888**
Main and internal workflows

@vpchung let's work on this branch.  Here are some todos:

- [x] The tool `determine_queue.cwl` needs some work - it needs to actually determine which internal queue to submit the submission to. Pulling out to separate issue #4 
- [x] start setting up the infra.  We should create 1 t2.medium and 1 smaller GPU instance for now.  Each instance will have an orchestrator.
      - The t2.medium will run the main workflow, run 1 submission at a time for now
      - The smaller GPU instance will run 1 submission at at time and run the internal workflow
      - Each internal queue we add will need its own GPU instance and will follow the configuration done for the above GPU instance.
- [x] Create subset of data and add another run_docker step to the internal workflow (`testing-phase-workflow.cwl`)
- [x] `run_docker.py` needs to support GPUs. 
- [x] No need to modify annotations.  The scores should be annotated in the same annotation keys
- [x] Hard code input directory path:  https://github.com/Sage-Bionetworks-Challenges/brats-dream-challenge-infra/issues/6

### CODE REVIEW

**Comment by vpchung** [comment]
Got around to configuring a GPU instance; [notes here](https://docs.google.com/document/d/1wyjYoxF4sfGP4uLdY2Z0aod-yNiT5lwovhIJT6m_xcw/edit?usp=sharing).  Tomorrow, I'll start modifying and testing the workflow with the testing instance.

File: testing-phase-workflow.cwl [hunk 1]
```diff
@@ -159,92 +131,72 @@ steps:
...
         source: "#submitterUploadSynId"
       - id: synapse_config
         source: "#synapseConfig"
-      # OPTIONAL: set `default` to `false` if log file should not be uploaded to Synapse
-      - id: store
-        default: true
-      # TODO: replace `valueFrom` with the absolute path to the data directory to be mounted
       - id: input_dir
-        valueFrom: "/tmp"
+        source: "#get_evaluation_config/dataset_path"
```
Comment (vpchung) [inline testing-phase-workflow.cwl:142]: Workflow currently missing this step + CWL.

**Comment by vpchung** [comment]
Update:
- [ ] **The tool determine_queue.cwl needs some work**
Haven't had time to look at this yet, but will be the first thing I check on Friday.

- [x] **start setting up the infra.**
Done, except I launched a t3.medium, not t2.medium.

- [x] **Create subset of data and add another run_docker step**
Done but I haven't tested yet.  Plus, I think we are missing a step in the workflow in order to set some of the params.  Unless I can just hardcode the values in?

- [x] **run_docker.py needs to support GPUs.**
Done, and I _think_ it's working as expected??  Will know more when I use one of last year's models as a test.

File: testing-phase-workflow.cwl (hunk 1 above)
Comment (thomasyu888) [reply testing-phase-workflow.cwl:171]: No need for this, just use the actual input_directory path here.

**Pull request merged**